woocommerce/woocommerce
# [e2e tests] Add tests for product images

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Adds new tests covering the product image functionality.

Closes https://github.com/woocommerce/woocommerce-quality/issues/615 .

- Added a new spec file `tests/merchant/product-images.spec.js` with 5 tests.
- Added 3 images in test data to be used.
- The 3 images are uploaded in the environment setup via the `test-env-setup.sh` script, so they are available on the Media Library.
- The fixtures that are creating products with images are using images from `http://demo.woothemes.com`. I wanted to use one of the 3 test images, but that's not possible, because the image url needs to be public, otherwise the validation of the image url fails in the api call. Unless the test site is publicly reachable, I see no way to use the api to create the product with images. This is not ideal because we do no control the images in `http://demo.woothemes.com`, but I don't expect them to change that often, and I think it's good enough for now to get this PR over the line.


Unrelated to this task:

- Added `repeatEach` in Playwright's config, with default value of 1 (nothing changes) and option to override the value via the `REPEAT_EACH` env variable. This can be helpful to easily run tests many times to check for flakiness. I used it to run this new spec 100 times in a single run and had no failures.
- Simplified the fixtures in `product-delete.spec.js`. It's very similar to the fixture used in this PR's spec.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. You can run the new spec locally with
`pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/product-images.spec.js`. Make sure you reset the e2e environment before (`pnpm env:restart`) so the images used by the new tests are imported.
2. Make sure the new tests pass in CI



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
